### PR TITLE
fix: Render bridges correctly for v2 on sysconfig with set-name

### PR DESCRIPTION
**Pull request by TheRealFalcon**
## Proposed Commit Message

```
fix: Render bridges correctly for v2 on sysconfig with set-name

When listing interfaces in v2 format, we should expect to be able to
reference other interfaces using the name in the configuration, not
the name the interface will eventually take. This was broken when
using `set-name`.

To fix this, we now store the configuration id alongside the eventual
name, and reference that instead of the name.

Fixes GH-5574
```

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Comment by TheRealFalcon** [comment]
@xiachen-rh , does this solve the issue that you raised in #5574 ?

**Comment by xiachen-rh** [comment]
Sorry I just notice this, I will test it ASAP and let you know the result.

**Comment by xiachen-rh** [comment]
@TheRealFalcon Hi James, the fix works for sysconfig renderer.  However, we still have an issue for NM renderer, so I filed a new issue https://github.com/canonical/cloud-init/issues/5717 to track it.

**Comment by TheRealFalcon** [comment]
@xiachen-rh , thanks for checking and filing that issue. We can follow up on that separately.

@holmanb , this is ready for review.

**Comment by ani-sinha** [comment]
@TheRealFalcon @holmanb waiting for this to get merged. Then https://github.com/canonical/cloud-init/pull/5740 .

File: cloudinit/net/eni.py [hunk 1]
```diff
@@ -421,6 +422,11 @@ def _render_route(self, route, indent=""):
         return content
 
     def _render_iface(self, iface, render_hwaddress=False):
+        iface = copy.deepcopy(iface)
+
+        # Remove irrelevant keys
+        with suppress(KeyError):
```
Comment (holmanb) [inline cloudinit/net/eni.py:428]: I like using `suppress`, but in this case it doesn't appear to add value. Wouldn't something like this suffice?
```python
if iface.get("config_id"):
    iface.pop("config_id")
```

File: cloudinit/net/network_state.py [hunk 2]
```diff
@@ -800,26 +803,14 @@ def handle_wifis(self, command):
...
-                set_name_iface = dev_cfg.get("set-name")
-                if set_name_iface:
-                    iface = set_name_iface
             if "nameservers" in dev_cfg:
                 search = dev_cfg.get("nameservers").get("search", [])
                 dns = dev_cfg.get("nameservers").get("addresses", [])
                 name_cmd = {"type": "nameserver"}
                 if len(search) > 0:
-                    name_cmd.update({"search": search})
+                    name_cmd["search"] = search
```
Comment (holmanb) [inline cloudinit/net/network_state.py:811]: We could similarly exploit the falsyness of `search` and `dns` (no need to check that `len() > 0`) and also remove the default values in `.get("search", [])` and `.get("addresses", [])`, if you want.

**Review by holmanb — APPROVED** [review]
@TheRealFalcon Thanks for this! I left a couple of nitpicks, take them or leave them. Either way it looks good to me.

**Pull request merged**